rtgui/flatfield.cc

```diff
@@ -403,3 +403,21 @@ void FlatField::setShortcutPath(const Glib::ustring& path)
 
     } catch (Glib::Error&) {}
 }
+
+void FlatField::flatFieldAutoClipValueChanged(int n)
+{
+    struct Data {
+        FlatField *me;
+        int n;
+    };
+    const auto func = [](gpointer data) -> gboolean {
+        Data *d = static_cast<Data *>(data);
+        FlatField *me = d->me;
+        me->disableListener();
+        me->flatFieldClipControl->setValue (d->n);
+        me->enableListener();
+        return FALSE;
```

Comment:
@heckflosse Ingo, you should delete `d` as a last step, as the `IdleRegister` won't touch it. It will still leak in case `func` wasn't called due to early destruction, and this is something I have to think about. Maybe an optional deleter (as a `std::function<>`) would be the right approach.

Furthermore, you should call `idle_register.destroy()` early in `~FlatField()`. Its automatic destruction is more like a safety net, other members of the containing class could already be gone.

Best,
Flössie

Reply:
Done